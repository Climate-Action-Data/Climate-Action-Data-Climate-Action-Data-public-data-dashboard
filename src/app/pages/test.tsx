'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test