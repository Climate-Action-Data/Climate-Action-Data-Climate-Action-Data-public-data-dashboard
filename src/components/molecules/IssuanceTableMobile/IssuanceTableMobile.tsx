import { Issuance, IssuanceUnit } from '@/@types/ProjectDetails'
import { IssuanceTableMobileItem } from '@/components/atoms/IssuanceTableMobileItem/IssuanceTableMobileItem'
import { NoDataScreen } from '@/components/atoms/NoDataScreen/NoDataScreen'
import { RetirementMobileTable } from '@/components/atoms/RetirementMobileTable/RetirementMobileTable'
import { Accordion, Box, Button, Flex, Heading, Modal, ModalBody, ModalContent, ModalFooter, ModalHeader, ModalOverlay, useDisclosure } from '@chakra-ui/react'
import { useState } from 'react'
import { useTranslation } from 'react-i18next'

interface IssuanceTableMobileProps {
  issuances: Issuance[]
}

export const IssuanceTableMobile = (props: IssuanceTableMobileProps) => {
  const { issuances } = props
  const [retirements, setRetirements] = useState<IssuanceUnit[]>([])
  const { isOpen, onOpen, onClose } = useDisclosure()
  const { t } = useTranslation(`projectDetails`)

  const handleIssuanceClick = (retirements: IssuanceUnit[]) => {
    setRetirements(retirements)
    onOpen()
  }

  const renderAccordionItems = () => {
    return issuances.map((issuance) => (
      <IssuanceTableMobileItem onClick={(retirements) => handleIssuanceClick(retirements)} issuance={issuance} key={`issuance-table-mobile-item-${issuance.id}`} />
    ))
  }

  const renderHeadingWithUnderling = (text: string) => {
    return (
      <Box minH="56px" borderBottom="2px solid" borderBottomColor="lightGray.700">
        <Flex padding="0 24px" alignItems="center" height="56px">
          <Heading variant={`aeonik`} color="lightGray.700" textTransform="uppercase" as="h3" size="16px">
            {text}
          </Heading>
        </Flex>
      </Box>
    )
  }

  return (
    <Box w="100%">
      {renderHeadingWithUnderling(t(`issuances.title`))}
      <Accordion gap="16px" padding="0px 0px 16px 0px" allowToggle>
        {renderAccordionItems()}
      </Accordion>
      <Modal isCentered isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent marginX="12px" minH="503px" maxH="503px">
          <ModalHeader p={0}>{renderHeadingWithUnderling(t(`issuances.retirememts`))}</ModalHeader>
          <ModalBody overflowY="scroll" p="12px" paddingBottom="20px">
            <Flex width="100%" gap="16px" flexDirection="column" padding="0px">
              {retirements.length > 0 ? <RetirementMobileTable retirements={retirements} /> : <NoDataScreen message={t(`issuances.noRetirementData`)} />}
            </Flex>
          </ModalBody>

          <ModalFooter>
            <Button variant="modalClose" onClick={onClose}>
              Close
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Box>
  )
}
