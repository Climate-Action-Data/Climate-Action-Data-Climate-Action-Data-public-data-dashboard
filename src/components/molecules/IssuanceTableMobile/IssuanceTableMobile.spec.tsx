import { render, screen, fireEvent } from '@testing-library/react'
import { IssuanceTableMobile } from './IssuanceTableMobile'
import { PROJECT_DETAIL } from '@/test/TestOvermindMockData'

describe(`IssuanceTableMobile`, () => {
  it(`renders correctly issuances`, () => {
    const { container } = render(<IssuanceTableMobile issuances={PROJECT_DETAIL.issuances} />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly with empty array`, () => {
    const { container } = render(<IssuanceTableMobile issuances={[]} />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly with empty array`, () => {
    const { container } = render(<IssuanceTableMobile issuances={PROJECT_DETAIL.issuances} />)
    const items = screen.getAllByTestId(`issuance-mobile-item-click`)
    expect(items.length).toBeGreaterThan(0)
    fireEvent.click(items[0])
    expect(container).toMatchSnapshot()
  })
})
