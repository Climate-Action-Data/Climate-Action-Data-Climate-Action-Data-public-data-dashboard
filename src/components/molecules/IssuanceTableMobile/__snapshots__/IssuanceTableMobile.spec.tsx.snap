// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuanceTableMobile renders correctly issuances 1`] = `
<div>
  <div
    class="css-8atqhb"
  >
    <div
      class="css-1bnwekn"
    >
      <div
        class="css-1i18okm"
      >
        <h3
          class="chakra-heading css-zux3ex"
        >
          Issuances
        </h3>
      </div>
    </div>
    <div
      class="chakra-accordion css-1cvb60n"
    >
      <div
        class="chakra-accordion__item css-1obe73b"
      >
        <h2>
          <button
            aria-controls="accordion-panel-:r1:"
            aria-expanded="false"
            class="chakra-accordion__button css-1kfav1u"
            data-index="0"
            data-testid="expand-accordion"
            id="accordion-button-:r1:"
            type="button"
          >
            <div
              class="css-13buc9q"
            >
              <div
                class="css-xrub82"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Vintage
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    2010
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Quantity
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    191
                  </p>
                </div>
              </div>
              <svg
                class="chakra-icon css-n059si"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.0045 15.0135C11.8904 15.0135 11.7798 14.9942 11.6728 14.9558C11.5657 14.9173 11.4654 14.8513 11.3718 14.7577L6.86221 10.2481C6.71734 10.1032 6.64747 9.93014 6.65261 9.72886C6.65772 9.52758 6.73272 9.3545 6.87761 9.20964C7.02247 9.06477 7.19811 8.99234 7.40451 8.99234C7.61091 8.99234 7.78654 9.06477 7.93141 9.20964L12.0045 13.2827L16.093 9.19426C16.2378 9.04939 16.4109 8.97952 16.6122 8.98464C16.8135 8.98977 16.9865 9.06477 17.1314 9.20964C17.2763 9.3545 17.3487 9.53015 17.3487 9.73656C17.3487 9.94296 17.2763 10.1186 17.1314 10.2635L12.6372 14.7577C12.5436 14.8513 12.4449 14.9173 12.341 14.9558C12.2372 14.9942 12.125 15.0135 12.0045 15.0135Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r1:"
            class="chakra-accordion__panel css-old1by"
            id="accordion-panel-:r1:"
            role="region"
          >
            <div
              class="css-1ow11qq"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Units Available
                </p>
                <p
                  class="chakra-text css-0"
                >
                  191
                </p>
              </div>
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Verification Report Date
                </p>
                <p
                  class="chakra-text css-0"
                >
                  2020/03/04
                </p>
              </div>
            </div>
            <div
              class="css-gl17gz"
            >
              <a
                class="chakra-link css-0"
                data-testid="expand-more-details"
                href="/issuance?id=00cfede2-5701-4237-b7cf-8b962ba02aa5"
              >
                More Details
              </a>
              <button
                class="chakra-button css-4xx2wk"
                data-testid="issuance-mobile-item-click"
                type="button"
              >
                View Retirements
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-accordion__item css-1obe73b"
      >
        <h2>
          <button
            aria-controls="accordion-panel-:r3:"
            aria-expanded="false"
            class="chakra-accordion__button css-1kfav1u"
            data-index="1"
            data-testid="expand-accordion"
            id="accordion-button-:r3:"
            type="button"
          >
            <div
              class="css-13buc9q"
            >
              <div
                class="css-xrub82"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Vintage
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    2015
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Quantity
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    1,102
                  </p>
                </div>
              </div>
              <svg
                class="chakra-icon css-n059si"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.0045 15.0135C11.8904 15.0135 11.7798 14.9942 11.6728 14.9558C11.5657 14.9173 11.4654 14.8513 11.3718 14.7577L6.86221 10.2481C6.71734 10.1032 6.64747 9.93014 6.65261 9.72886C6.65772 9.52758 6.73272 9.3545 6.87761 9.20964C7.02247 9.06477 7.19811 8.99234 7.40451 8.99234C7.61091 8.99234 7.78654 9.06477 7.93141 9.20964L12.0045 13.2827L16.093 9.19426C16.2378 9.04939 16.4109 8.97952 16.6122 8.98464C16.8135 8.98977 16.9865 9.06477 17.1314 9.20964C17.2763 9.3545 17.3487 9.53015 17.3487 9.73656C17.3487 9.94296 17.2763 10.1186 17.1314 10.2635L12.6372 14.7577C12.5436 14.8513 12.4449 14.9173 12.341 14.9558C12.2372 14.9942 12.125 15.0135 12.0045 15.0135Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r3:"
            class="chakra-accordion__panel css-old1by"
            id="accordion-panel-:r3:"
            role="region"
          >
            <div
              class="css-1ow11qq"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Units Available
                </p>
                <p
                  class="chakra-text css-0"
                >
                  0
                </p>
              </div>
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Verification Report Date
                </p>
                <p
                  class="chakra-text css-0"
                >
                  2020/03/04
                </p>
              </div>
            </div>
            <div
              class="css-gl17gz"
            >
              <a
                class="chakra-link css-0"
                data-testid="expand-more-details"
                href="/issuance?id=136cadbc-83a5-4c43-a172-baf7770796fc"
              >
                More Details
              </a>
              <button
                class="chakra-button css-4xx2wk"
                data-testid="issuance-mobile-item-click"
                type="button"
              >
                View Retirements
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`IssuanceTableMobile renders correctly with empty array 1`] = `
<div>
  <div
    class="css-8atqhb"
  >
    <div
      class="css-1bnwekn"
    >
      <div
        class="css-1i18okm"
      >
        <h3
          class="chakra-heading css-zux3ex"
        >
          Issuances
        </h3>
      </div>
    </div>
    <div
      class="chakra-accordion css-1cvb60n"
    />
  </div>
</div>
`;

exports[`IssuanceTableMobile renders correctly with empty array 2`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <div
    class="css-8atqhb"
  >
    <div
      class="css-1bnwekn"
    >
      <div
        class="css-1i18okm"
      >
        <h3
          class="chakra-heading css-zux3ex"
        >
          Issuances
        </h3>
      </div>
    </div>
    <div
      class="chakra-accordion css-1cvb60n"
    >
      <div
        class="chakra-accordion__item css-1obe73b"
      >
        <h2>
          <button
            aria-controls="accordion-panel-:r9:"
            aria-expanded="false"
            class="chakra-accordion__button css-1kfav1u"
            data-index="0"
            data-testid="expand-accordion"
            id="accordion-button-:r9:"
            type="button"
          >
            <div
              class="css-13buc9q"
            >
              <div
                class="css-xrub82"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Vintage
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    2010
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Quantity
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    191
                  </p>
                </div>
              </div>
              <svg
                class="chakra-icon css-n059si"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.0045 15.0135C11.8904 15.0135 11.7798 14.9942 11.6728 14.9558C11.5657 14.9173 11.4654 14.8513 11.3718 14.7577L6.86221 10.2481C6.71734 10.1032 6.64747 9.93014 6.65261 9.72886C6.65772 9.52758 6.73272 9.3545 6.87761 9.20964C7.02247 9.06477 7.19811 8.99234 7.40451 8.99234C7.61091 8.99234 7.78654 9.06477 7.93141 9.20964L12.0045 13.2827L16.093 9.19426C16.2378 9.04939 16.4109 8.97952 16.6122 8.98464C16.8135 8.98977 16.9865 9.06477 17.1314 9.20964C17.2763 9.3545 17.3487 9.53015 17.3487 9.73656C17.3487 9.94296 17.2763 10.1186 17.1314 10.2635L12.6372 14.7577C12.5436 14.8513 12.4449 14.9173 12.341 14.9558C12.2372 14.9942 12.125 15.0135 12.0045 15.0135Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r9:"
            class="chakra-accordion__panel css-old1by"
            id="accordion-panel-:r9:"
            role="region"
          >
            <div
              class="css-1ow11qq"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Units Available
                </p>
                <p
                  class="chakra-text css-0"
                >
                  191
                </p>
              </div>
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Verification Report Date
                </p>
                <p
                  class="chakra-text css-0"
                >
                  2020/03/04
                </p>
              </div>
            </div>
            <div
              class="css-gl17gz"
            >
              <a
                class="chakra-link css-0"
                data-testid="expand-more-details"
                href="/issuance?id=00cfede2-5701-4237-b7cf-8b962ba02aa5"
              >
                More Details
              </a>
              <button
                class="chakra-button css-4xx2wk"
                data-testid="issuance-mobile-item-click"
                type="button"
              >
                View Retirements
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-accordion__item css-1obe73b"
      >
        <h2>
          <button
            aria-controls="accordion-panel-:rb:"
            aria-expanded="false"
            class="chakra-accordion__button css-1kfav1u"
            data-index="1"
            data-testid="expand-accordion"
            id="accordion-button-:rb:"
            type="button"
          >
            <div
              class="css-13buc9q"
            >
              <div
                class="css-xrub82"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Vintage
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    2015
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Quantity
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    1,102
                  </p>
                </div>
              </div>
              <svg
                class="chakra-icon css-n059si"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.0045 15.0135C11.8904 15.0135 11.7798 14.9942 11.6728 14.9558C11.5657 14.9173 11.4654 14.8513 11.3718 14.7577L6.86221 10.2481C6.71734 10.1032 6.64747 9.93014 6.65261 9.72886C6.65772 9.52758 6.73272 9.3545 6.87761 9.20964C7.02247 9.06477 7.19811 8.99234 7.40451 8.99234C7.61091 8.99234 7.78654 9.06477 7.93141 9.20964L12.0045 13.2827L16.093 9.19426C16.2378 9.04939 16.4109 8.97952 16.6122 8.98464C16.8135 8.98977 16.9865 9.06477 17.1314 9.20964C17.2763 9.3545 17.3487 9.53015 17.3487 9.73656C17.3487 9.94296 17.2763 10.1186 17.1314 10.2635L12.6372 14.7577C12.5436 14.8513 12.4449 14.9173 12.341 14.9558C12.2372 14.9942 12.125 15.0135 12.0045 15.0135Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:rb:"
            class="chakra-accordion__panel css-old1by"
            id="accordion-panel-:rb:"
            role="region"
          >
            <div
              class="css-1ow11qq"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Units Available
                </p>
                <p
                  class="chakra-text css-0"
                >
                  0
                </p>
              </div>
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-uwkl65"
                >
                  Verification Report Date
                </p>
                <p
                  class="chakra-text css-0"
                >
                  2020/03/04
                </p>
              </div>
            </div>
            <div
              class="css-gl17gz"
            >
              <a
                class="chakra-link css-0"
                data-testid="expand-more-details"
                href="/issuance?id=136cadbc-83a5-4c43-a172-baf7770796fc"
              >
                More Details
              </a>
              <button
                class="chakra-button css-4xx2wk"
                data-testid="issuance-mobile-item-click"
                type="button"
              >
                View Retirements
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
